sympy/physics/mechanics/joint.py

```diff
@@ -224,45 +269,101 @@ def _orient_frames(self):
 
     @abstractmethod
     def _set_angular_velocity(self):
+        """Set angular velocity of the joint related frames."""
         pass
 
     @abstractmethod
     def _set_linear_velocity(self):
+        """Set velocity of related points to the joint."""
         pass
 
+    @staticmethod
+    def _to_vector(matrix, frame):
+        """Converts a matrix to a vector in the given frame."""
+        v = Vector(0)
+        for xi, vi in zip(matrix, frame):
+            v += xi * vi
+        return v
+
+    @staticmethod
+    def _axis(ax, *frames):
+        """Check whether an axis is fixed in one of the frames."""
+        if ax is None:
+            ax = frames[0].x
+            return ax
+        if not isinstance(ax, Vector):
+            raise TypeError("Axis must be a Vector.")
+        ref_frame = None  # Find a body in which the axis can be expressed
+        for frame in frames:
+            try:
+                ax.to_matrix(frame)
+                ref_frame = frame
+                break
+            except ValueError:
+                pass
+        if ref_frame is None:
+            raise ValueError("Axis cannot be expressed in one of the body's "
+                             "frames.")
+        if not ax.dt(ref_frame) == 0:
+            raise ValueError('Axis cannot be time-varying when viewed from the '
+                             'associated body.')
+        return ax
+
+    def _express_axis(self, axis, frame):
+        """Helper function to get an axis expressed in a specified frame."""
+        try:
+            ax_mat = axis.to_matrix(self.parent_interframe)
+        except ValueError:
+            ax_mat = axis.to_matrix(self.child_interframe)
+        try:
+            self.parent_interframe.dcm(frame)  # Check if connected
+            int_frame = self.parent_interframe
```

Comment:
Move to separate `else` in try-except block as this line won't be the one throwing the intended `ValueError`.